### PR TITLE
[BUGFIX] [MER-713] Validation long product title

### PR DESCRIPTION
**Pull request by simonchoxx**
[MER-713](https://eliterate.atlassian.net/browse/MER-713)

### CODE REVIEW

File: test/oli_web/live/sections/edit_live_test.exs [hunk 1]
```diff
@@ -196,5 +196,46 @@ defmodule OliWeb.Sections.EditLiveTest do
              |> element("#section_display_curriculum_item_numbering")
              |> render() =~ "checked"
     end
+
+    test "update section with a long title shows an error alert", %{conn: conn, section: section} do
+      {:ok, view, _html} = live(conn, "/authoring/products/#{section.slug}")
```
Comment (mcass19) [inline test/oli_web/live/sections/edit_live_test.exs:201]: Would be great to not write the explicit path but instead use the route helpers, wdyt?

File: test/oli_web/live/sections/edit_live_test.exs [hunk 2]
```diff
@@ -196,5 +196,46 @@ defmodule OliWeb.Sections.EditLiveTest do
...
+      view
+      |> element("form[phx-submit=\"save\"")
+      |> render_submit(%{section: %{title: long_title}})
+
+      assert view
+             |> element("div.alert.alert-danger")
+             |> render() =~
+               "Couldn&#39;t update product title"
+
+      assert view
```
Comment (mcass19) [inline test/oli_web/live/sections/edit_live_test.exs:215]: I think this assert is ok, but what do you think about testing that the actual product was not updated?

File: test/oli_web/live/sections/edit_live_test.exs [hunk 3]
```diff
@@ -196,5 +196,46 @@ defmodule OliWeb.Sections.EditLiveTest do
...
+             |> element("#section_title")
+             |> render() =~ long_title
+
+      assert has_element?(view, "span", "Title should be at most 255 character(s)")
+    end
+
+    test "update section with a valid title shows an info alert", %{conn: conn, section: section} do
+      {:ok, view, _html} = live(conn, "/authoring/products/#{section.slug}")
```
Comment (mcass19) [inline test/oli_web/live/sections/edit_live_test.exs:223]: Same as above

File: test/oli_web/live/sections/edit_live_test.exs [hunk 4]
```diff
@@ -196,5 +196,46 @@ defmodule OliWeb.Sections.EditLiveTest do
...
+      view
+      |> element("form[phx-submit=\"save\"")
+      |> render_submit(%{section: %{title: valid_title}})
+
+      assert view
+             |> element("div.alert.alert-info")
+             |> render() =~
+               "Product changes saved"
+
+      assert view
```
Comment (mcass19) [inline test/oli_web/live/sections/edit_live_test.exs:236]: Same as above, would be great to test that the product was updated with `Valid title`

**Review by mcass19 — COMMENTED** [review]
Great job 🚀 🚀 🚀 

Just left some really minor things if you want to consider them, but this is awesome!!!

File: test/oli_web/live/sections/edit_live_test.exs (hunk 1 above)
Comment (simonchoxx) [reply test/oli_web/live/sections/edit_live_test.exs]: Sure. Perfect! I change it!

File: test/oli_web/live/sections/edit_live_test.exs (hunk 3 above)
Comment (simonchoxx) [reply test/oli_web/live/sections/edit_live_test.exs]: Sure. Perfect! I change it!

File: test/oli_web/live/sections/edit_live_test.exs (hunk 2 above)
Comment (simonchoxx) [reply test/oli_web/live/sections/edit_live_test.exs:219]: I think that is an excellent idea!

**Review by mcass19 — APPROVED** [review]

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**